src/esp/assets/ResourceManager.cpp

```diff
@@ -2056,7 +2080,12 @@ void ResourceManager::addComponent(
   // Add a drawable if the object has a mesh and the mesh is loaded
   if (meshIDLocal != ID_UNDEFINED) {
     const int meshID = metaData.meshIndex.first + meshIDLocal;
-    Magnum::GL::Mesh& mesh = *meshes_.at(meshID)->getMagnumGLMesh();
+    Magnum::GL::Mesh* mesh = meshes_.at(meshID)->getMagnumGLMesh();
+    if (getCreateRenderer()) {
+      CORRADE_INTERNAL_ASSERT(mesh);
+    } else {
+      CORRADE_INTERNAL_ASSERT(!mesh);
```

Comment:
Use CORRADE_ASSERT instead in order to output the error message.
e.g., 
"ResourceManager::addComponent(): Simulator is not created with createRenderer is enabled. Check your simulator config."

It will significantly save developer's debugging time.

Reply:
I think you misunderstand the assert. For the else case, this assert is verifying that we haven't created a GL mesh, as expected. The code elsewhere that avoids calling `uploadBuffersToGPU` ensures this. A user isn't going to hit this assert. This is an internal assert.

With that said, there's no harm in changing both of these to `CORRADE_ASSERT` and adding some explanation text. For the second one, it would be something like "encountered unexpected GL mesh with createRenderer==false".